src/components/GameboardOnline/GameboardOnline.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect } from "react";
+import PropTypes from "prop-types";
+import { getAuth } from "firebase/auth";
+import { gameIsOver, updateGameOnline } from "../../services/game.service";
+import { PlayCard } from "../PlayCard/PlayCard";
+
+const turnIsAllowed = (turn) => turn < 2;
+
+const endOfGame = (board) => {
+  return board.every((card) => card.cleared);
+};
+
+const getNextPlayer = ({ players, currentPlayer }) => {
+  const playerArray = Object.keys(players);
+  const currentPlayerIndex = playerArray.indexOf(currentPlayer);
+  const nextPlayerIndex = (currentPlayerIndex + 1) % playerArray.length;
+  return playerArray[nextPlayerIndex];
+};
+
+const revealCard = ({ cardID, board }) => {
+  const newBoard = board.map((card) => {
+    if (cardID === card.id) {
+      return {
+        ...card,
+        reveal: true,
+      };
+    }
+    return card;
+  });
+  return newBoard;
+};
+
+const unrevealCards = (board) => {
+  return board.map((card) => (card.reveal ? { ...card, reveal: false } : card));
+};
+
+const revealedCardsMatch = (board) => {
+  const revealedCards = board.filter((card) => card.reveal);
+  return (
+    revealedCards.map((card) => card.id).reduce((acc, cur) => acc + cur) === 0
+  );
+};
+const clearCards = (board) => {
+  return board.map((card) => {
+    return card.reveal ? { ...card, cleared: true, reveal: false } : card;
+  });
+};
```

Comment:
Diese ganzen kleinen FUnktionen können doch auch offline genutzt werden. Ich weiß du hast die neugeschrieben. Kann man später in einen services oder controller auslagern hier weniger code zu haben und das offline und online die gleichen Funktionen nutzen.

Reply:
find ich gut, ich mach mir ne Notiz in Trello